File: JavaScript/hamming.js
const compute = (a, b) => {
  if(a.length !== b.length) { throw new Error('strands must be of equal length'); }

  let differences = 0;

  for(let i = 0; i < a.length; i++) {
    if(a[i] != b[i]) { 
      differences++; 
    }
  }

  return differences;
};

console.log(compute("GAGCCTACTAACGGGAT", "CATCGTAATGACGGCCT"))

File: C#/Hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length) 
        {
            throw new ArgumentException("Strands must be of equal length"); 
        }

        int differences = 0;

        for (int i = 0; i < firstStrand.Length; i++)
        {
            if (firstStrand[i] != secondStrand[i])
            {
                differences++;
            }
        }

        return differences;
    }
}